docs/content/v0-upgrade.md

```diff
@@ -0,0 +1,246 @@
+
+#### How to Run
+
+- Policies are authored to be v0.x compatible.
+- v0.x producers and v0.x consumers are run as is.
+
+### Next
+
+Downgrade consumers to v0.x ([Scenario 1](#scenario-1)) until all consumers are
+v0. It's also possible to upgrade producers to v1.0 and continue the upgrade
+from that point. Generally, it's recommended to upgrade producers first, however
+depending on your existing OPAs v1.0 consumers deployments, you may prefer to
+upgrade all your producers to v1.0 rather than to downgrade consumers.
+
+### Scenario 5
+
+Mixed versions of OPA are being used for both bundle production and consumption.
+
+#### Pre-requisites
+
+As users have a mix of bundle producers, they must have control over the runtime
+options for the producers to set `--v0-compatible`. Users must also have control
+over their v1 consumers to set the `--v0-compatible` flag. Both these conditions
+must be met for the upgrade to proceed.
+
+#### How to Run
+
+- Policies are authored to be v0.x compatible.
+- v1.0 consumers are run with `--v0-compatible`.
+
+### Next
+
+Please gradually upgrade producers to v1.0 ([Scenario 8](#scenario-8)) until all
+producers are v1.0 ([Scenario 9](#scenario-9)).
+
+### Scenario 6
+
+All consumers can be run without flags, as the bundle will contain
+attributes to inform v1.0 OPAs to accept v0.x modules.
+
+#### How to Run
+
+- Policies are authored to be v0.x compatible.
+- v0.x consumers are run as-is. Bundles will contain v0.x policies
+- v1.0 consumers are run as-is. Bundles will contain rego version attribute, so v0.x modules are accepted.
+
+#### Pre-requisites
+
+If users cannot set their OPA v1 producers to use `--v0-compatible` to be
+compatible with their v0.x consumers, then this upgrade path is blocked.
+
+#### Next
+
+Running exclusively v1.0 producers and consumers, ([Scenario 9](#scenario-9)), is
+the next and final step.
+
+### Scenario 7
+
+All consumers are v1.0, but producers are v0.x. This scenario might occur when
+OPAs used for evaluation are upgraded before the policy bundling system.
+
+#### Pre-requisites
+
+If v1.0 consumers cannot be run with `--v0-compatible`, when loading v0 consumer
+generated bundle, the bundles cannot include Rego version attribute. This means
+the upgrade path is blocked until either the consumers can create bundles with a
+Rego version or the `--v0-compatible` flag is available for producers.
+
+#### How to Run
+
+- Policies are authored to be v0.x compatible.
+- v1.0 consumers are run with `--v0-compatible`
+
+#### Next
+
+Upgrade producers to v1.0 ([Scenario 8](#scenario-8)) until all producers are v1.0
+([Scenario 9](#scenario-9)).
+
+### Scenario 8
+
+All consumers are v1.0, but producers are a mix of v0.x and v1.0.
+
+#### How to Run
+
+- Policies are authored to be v0.x compatible.
+- v1.0 consumers are run with `--v0-compatible`
+- v1.0 producers are run with `--v0-compatible`
+- `rego.v0` imports cannot be used; rejected by v0.x bundle producers.
```

Comment:
Depends on how old v0.x versions are.

Reply:
We don't have a `rego.v0` import. Updated.